src/app-layout/visual-refresh/trigger-button.tsx

```diff
@@ -32,38 +69,134 @@ function TriggerButton(
+
   return (
-    <div className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}>
-      <button
-        aria-expanded={ariaExpanded}
-        aria-controls={ariaControls}
-        aria-haspopup={true}
-        aria-label={ariaLabel}
-        className={clsx(
-          styles.trigger,
-          {
-            [styles.selected]: selected,
-            [styles.badge]: badge,
-          },
-          className
-        )}
-        onClick={onClick}
-        ref={ref as React.Ref<HTMLButtonElement>}
-        type="button"
-        data-testid={testId}
-      >
-        <span className={clsx(badge && styles['trigger-badge-wrapper'])}>
-          <Icon name={iconName} svg={iconSvg} />
-        </span>
-      </button>
-      {badge && <div className={styles.dot} />}
+    <div
+      ref={containerRef}
+      {...(hasTooltip
+        ? {
+            onPointerEnter: () => onShowTooltipSoft(true),
+            onPointerLeave: () => onShowTooltipSoft(false),
+            onFocus: ({ target, currentTarget }) => {
+              if (currentTarget === target) {
+                onShowTooltipHard(true);
+              }
+            },
+            onBlur: () => onShowTooltipHard(false),
+          }
+        : {})}
+      data-testid={`${testId ? `${testId}-wrapper` : 'awsui-app-layout-trigger-wrapper'}${hasTooltip ? '-with-possible-tooltip' : ''}`}
+      className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}
+    >
+      {isMobile ? (
+        <InternalButton
+          ariaExpanded={ariaExpanded}
+          ariaLabel={ariaLabel}
+          className={className}
+          disabled={disabled}
+          ref={ref}
+          formAction="none"
+          iconName={iconName}
+          iconSvg={iconSvg}
+          badge={badge}
+          onClick={onClick}
+          variant="icon"
+          __nativeAttributes={{
+            'aria-haspopup': true,
+            ...(testId
+              ? {
+                  'data-testid': testId,
+                }
+              : {}),
+          }}
+        />
+      ) : (
+        <>
+          <button
+            aria-expanded={ariaExpanded}
+            aria-controls={ariaControls}
+            aria-haspopup={true}
+            aria-label={ariaLabel}
+            disabled={disabled}
+            className={clsx(
+              styles.trigger,
+              {
+                [styles.selected]: selected,
+                [styles.badge]: badge,
+              },
+              className
+            )}
+            onClick={onClick}
+            ref={ref as Ref<HTMLButtonElement>}
+            type="button"
+            data-testid={testId}
+          >
+            <span className={clsx(badge && styles['trigger-badge-wrapper'])}>
+              {(iconName || iconSvg) && <Icon name={iconName} svg={iconSvg} />}
+            </span>
+          </button>
+          {badge && <div className={styles.dot} />}
```

Comment:
InternalButton has it's own badge rendering so a badge only needs to be added for desktop